web/components/landing-page-panel.tsx

```diff
@@ -1,64 +1,451 @@
-import Image from 'next/image'
-import { SparklesIcon } from '@heroicons/react/solid'
-
-import { Contract } from 'common/contract'
-
-import { Spacer } from './layout/spacer'
+import { UserCircleIcon } from '@heroicons/react/solid'
 import { firebaseLogin } from 'web/lib/firebase/users'
-import { ContractsGrid } from './contract/contracts-grid'
 import { Col } from './layout/col'
 import { Row } from './layout/row'
 import { withTracking } from 'web/lib/service/analytics'
 import { useTracking } from 'web/hooks/use-tracking'
+import { useIsMobile } from 'web/hooks/use-is-mobile'
+import SquiggleVerticalIcon from 'web/lib/icons/squiggle_vertical'
+import clsx from 'clsx'
+import { Button } from './buttons/button'
+import { useEffect, useState } from 'react'
+import SquiggleHorizontalIcon from 'web/lib/icons/squiggle_horizontal'
+import TypewriterComponent from 'typewriter-effect'
+import EquilateralLeftTriangle from 'web/lib/icons/equilateral-left-triangle'
+import EquilateralRightTriangle from 'web/lib/icons/equilateral-right-triangle'
+import CountUp from 'react-countup'
+
+export type PageNumber = 0 | 1 | 2
+
+export function getNextPageNumber(pageNumber: PageNumber): PageNumber {
+  switch (pageNumber) {
+    case 0:
+      return 1
+    case 1:
+      return 2
+    case 2:
+      return 0
+  }
+}
 
-export function LandingPagePanel(props: { hotContracts: Contract[] }) {
-  const { hotContracts } = props
+export function PaginationCircle(props: {
+  currentPageNumber: PageNumber
+  pageNumber: PageNumber
+  onClick: () => void
+}) {
+  const { currentPageNumber, pageNumber, onClick } = props
+  return (
+    <div
+      onClick={() => onClick()}
+      className={clsx(
+        'h-2 w-2 cursor-pointer rounded-full transition-colors',
+        currentPageNumber === pageNumber ? 'bg-white' : 'bg-indigo-400'
+      )}
+    />
+  )
+}
 
+export function LandingPagePanel() {
   useTracking('view landing page')
+  const isMobile = useIsMobile()
+  const desktop_height = 'h-60' //240px
+  const mobile_height = 'h-96'
+
+  const [isButtonHover, setIsButtonHovered] = useState(false)
+  const [pageNumber, setPageNumber] = useState<PageNumber>(1)
+  const [currTimeoutId, setCurrTimeoutId] = useState<
+    NodeJS.Timeout | undefined | null
+  >(null)
+
+  useEffect(() => {
+    if (currTimeoutId) {
+      clearTimeout(currTimeoutId)
+    }
+    const newTimeoutId = setTimeout(
+      () => setPageNumber(getNextPageNumber(pageNumber)),
+      5500
+    )
+    setCurrTimeoutId(newTimeoutId)
```

Comment:
I think a cleaner way to clear the timeout is to return a cleanup function for `useEffect` instead of saving `currTimeoutId` as state:

```
return () => clearTimeout(newTimeoutId)
```